src/tools/rbd/action/Snap.cc

```diff
@@ -100,11 +102,21 @@ int do_list_snaps(librbd::Image& image, Formatter *f, bool all_snaps, librados::
         s->id, &trash_original_name);
     }
 
+    r = image.snap_is_protected(s->name.c_str(), &snap_protected);
```

Comment:
I think you should use logic similar to the following:

```
bool snap_protected = false;
if (snap_namespace == RBD_SNAP_NAMESPACE_TYPE_USER) {
  r = image.snap_is_protected(s->name.c_str(), &snap_protected);
  if (r < 0) {
    std::cerr << "rbd: unable to retrieve snap protection" << std::endl;
    return r;
  }
}
```

... and then below in the JSON/XML formatted output you would use ```snap_protected ? "true" : "false"``` and in the table output you would use ```snap_protected ? "yes" : ""```

Reply:
Hi @dillaman , Thank you very much for your detailed explanation, I have changed it.